src/util-exception-policy.c

```diff
@@ -69,36 +81,86 @@ void ExceptionPolicyApply(Packet *p, enum ExceptionPolicy policy, enum PacketDro
                 DecodeSetNoPacketInspectionFlag(p);
                 break;
         }
+    } else {
+        switch (policy) {
+            case EXCEPTION_POLICY_NOT_SET:
+                break;
+            case EXCEPTION_POLICY_BYPASS_FLOW:
+                PacketBypassCallback(p);
+                /* fall through */
+            case EXCEPTION_POLICY_PASS_FLOW:
+                SCLogDebug("EXCEPTION_POLICY_PASS_FLOW");
+                if (p->flow) {
+                    p->flow->flags |= FLOW_ACTION_PASS;
+                    FlowSetNoPacketInspectionFlag(p->flow); // TODO util func
+                }
+                /* fall through */
+            case EXCEPTION_POLICY_PASS_PACKET:
+                SCLogDebug("EXCEPTION_POLICY_PASS_PACKET");
+                DecodeSetNoPayloadInspectionFlag(p);
+                DecodeSetNoPacketInspectionFlag(p);
+                break;
+            default:
+                break;
+        }
     }
     SCLogDebug("end");
 }
 
 enum ExceptionPolicy ExceptionPolicyParse(const char *option, const bool support_flow)
 {
-    enum ExceptionPolicy policy = EXCEPTION_POLICY_IGNORE;
+    enum ExceptionPolicy policy = EXCEPTION_POLICY_NOT_SET;
     const char *value_str = NULL;
+    bool is_ips_mode = EngineModeIsIPS();
     if ((ConfGet(option, &value_str)) == 1 && value_str != NULL) {
         if (strcmp(value_str, "drop-flow") == 0) {
-            policy = EXCEPTION_POLICY_DROP_FLOW;
-            SCLogConfig("%s: %s", option, value_str);
+            if (is_ips_mode) {
+                policy = EXCEPTION_POLICY_DROP_FLOW;
+                SCLogConfig("%s: %s", option, value_str);
+            } else {
+                policy = EXCEPTION_POLICY_NOT_SET;
+                SCLogConfig("%s: %s not a valid config in IDS mode. Setting it to disabled)",
+                        option, value_str);
+            }
         } else if (strcmp(value_str, "pass-flow") == 0) {
             policy = EXCEPTION_POLICY_PASS_FLOW;
             SCLogConfig("%s: %s", option, value_str);
         } else if (strcmp(value_str, "bypass") == 0) {
             policy = EXCEPTION_POLICY_BYPASS_FLOW;
             SCLogConfig("%s: %s", option, value_str);
         } else if (strcmp(value_str, "drop-packet") == 0) {
-            policy = EXCEPTION_POLICY_DROP_PACKET;
-            SCLogConfig("%s: %s", option, value_str);
+            if (is_ips_mode) {
+                policy = EXCEPTION_POLICY_DROP_PACKET;
+                SCLogConfig("%s: %s", option, value_str);
+            } else {
+                policy = EXCEPTION_POLICY_NOT_SET;
+                SCLogConfig("%s: %s not a valid config in IDS mode. Setting it to disabled)",
+                        option, value_str);
+            }
```

Comment:
I'm not happy with this solution, as it duplicates a lot of code. The alternative could be.... to have a separate function to parse the exception policy master switch?